File: 0162-find-peak-element/0162-find-peak-element.java
class Solution {
    public int findPeakElement(int[] nums) {
        if (nums.length == 1) return 0;
        for (int x = 1; x < nums.length - 1; x++) {
            if (nums[x] > nums[x - 1] && nums[x] > nums[x + 1]) return x;
        }
        return nums[0] > nums[1] ? 0 : nums.length - 1;
    }
}

File: 0162-find-peak-element/0162-find-peak-element.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        if len(nums) == 1 : return 0
        if nums[0] > nums[1] : return 0
        if nums[len(nums) - 1] > nums[len(nums) - 2] : return len(nums) - 1

        x, y = 0, len(nums) - 1
        while x <= y :
            mid = int(x + (y - x) / 2)
            if (nums[mid] > nums[mid + 1] and nums[mid] > nums[mid - 1]) : return mid
            elif (nums[mid] < nums[mid + 1] and nums[mid] > nums[mid - 1]) : x = mid + 1
            else : y = mid

        return 0 if nums[0] > nums[1] else len(nums) - 1